src/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/AssemblyName.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace System.Reflection.Metadata
+{
+    internal static class AssemblyNameUtils
+    {
+        internal static void SetHashAlgorithm(this AssemblyName assemblyName, AssemblyHashAlgorithm assemblyHashAlgorithm)
+        {
+            var hashAlgorithmMappings = new Dictionary<AssemblyHashAlgorithm, Configuration.Assemblies.AssemblyHashAlgorithm>()
+            {
+                [AssemblyHashAlgorithm.MD5] = Configuration.Assemblies.AssemblyHashAlgorithm.MD5,
+                [AssemblyHashAlgorithm.Sha1] = Configuration.Assemblies.AssemblyHashAlgorithm.SHA1,
+                [AssemblyHashAlgorithm.Sha256] = Configuration.Assemblies.AssemblyHashAlgorithm.SHA256,
+                [AssemblyHashAlgorithm.Sha384] = Configuration.Assemblies.AssemblyHashAlgorithm.SHA384,
+                [AssemblyHashAlgorithm.Sha512] = Configuration.Assemblies.AssemblyHashAlgorithm.SHA512
+            };
+
+            foreach (var hashAlgorithmItem in hashAlgorithmMappings)
+            {
+                bool hasValue = (assemblyHashAlgorithm & hashAlgorithmItem.Key) != 0;
+
+                if (hasValue)
+                {
+                    assemblyName.HashAlgorithm |= hashAlgorithmItem.Value;
+                }
+                else
+                {
+                    assemblyName.HashAlgorithm &= ~hashAlgorithmItem.Value;
+                }
+            }
+        }
+
+        internal static void SetFlags(this AssemblyName assemblyName, AssemblyFlags flags)
+        {
+            var flagMappings = new Dictionary<AssemblyFlags, AssemblyNameFlags>()
```

Comment:
This will also allocate a Dictionary on every call. It looks prettier but is not as efficient as looking at if the flag is set through an & operation and then set it if so. i.e

```cs
if ((AssemblyFlags.PublicKey & flags) != 0)
     assemblyName.Flags |= AssemblyNameFlags.PublickKey
```

But if we add more flags later we would need to change this code, so perhaps using a cast is better, just as suggested here: https://github.com/dotnet/corefx/pull/23396/files#r134070466